src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -362,32 +362,35 @@ class AuthenticationCompleteException : Exception { }
 
         string? GetPassword(string username)
         {
-            var passwd = Settings.Password;
-            if (passwd != null)
-                return passwd;
+            var password = Settings.Password;
+            if (password != null)
+                return password;
 
             // No password was provided. Attempt to pull the password from the pgpass file.
-            var matchingEntry = PgPassFile.Load(Settings.Passfile)?.GetFirstMatchingEntry(Host, Port, Settings.Database!, username);
-            if (matchingEntry != null)
+            var passFilePath = Settings.Passfile ?? PostgresEnvironment.PassFile ?? PostgresEnvironment.PassFileDefault;
```

Comment:
As with the other setting, if the user explicitly specifies a Passfile in connection string, and that file doesn't exist, we should throw, no? But not with the environment variable.

Reply:
Tracked by #2694.